test/docfx.Test/lib/JsonSchemaTest.cs

```diff
@@ -0,0 +1,58 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Linq;
+using Xunit;
+
+namespace Microsoft.Docs.Build
+{
+    public class JsonSchemaTest
+    {
+        [Theory]
+
+        // type validation
```

Comment:
we want to make sure all our test cases are consistent with Json .Net behaviors, can we copy some tests from there?

Reply:
sure in another PR

Reply:
Seems Json team is also working on this feature, with a total different way, we may want to hold this PR until we all agree to do that in this way.

cc @LowEntropyBody